### ISSUE TITLE
AST parser Error with svg inline

### ISSUE THREAD

**Issue by ryuran** [pocketjoso/penthouse#201]
# KO when 2 svg (first with parentheses) are on same line (minified css)
```css
.test1{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 289 300'%3E%3Cstyle%3Epath%7Bfill:rgb(0,0,; %7D%3C/style%3E%3C/svg%3E")}.test2{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 263 211' preserveAspectRatio=''%3E%3C/svg%3E")}
```
http://astexplorer.net/#/gist/f33d774445559ff546b48875302fbe30/2088fcff04d0be80b2b0bfec1eb1103a71ac2da8

# OK when it’s on 2 separates lines
```css
.test1{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 289 300'%3E%3Cstyle%3Epath%7Bfill:rgb(0,0,0)%3B %7D%3C/style%3E%3C/svg%3E")}
.test2{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 263 211' preserveAspectRatio=''%3E%3C/svg%3E")}
```
http://astexplorer.net/#/gist/90a203f59b5a24498fae41237c968d73/a711894d57e88ff50c28daa80c61ee2250a2ceb7

# OK with parentheses encoded
```css
.test1{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 289 300'%3E%3Cstyle%3Epath%7Bfill:rgb%280,0,0%29%3B %7D%3C/style%3E%3C/svg%3E")}.test2{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 263 211' preserveAspectRatio=''%3E%3C/svg%3E")}
```
http://astexplorer.net/#/gist/614652fb547b0988b92eb2d5f86693bf/32ce8bb7ac4e4588d2c132c78a67a0efb9e8ffe4

# KO when svg with semicolon
```css
.test1{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 289 300'%3E%3Cstyle%3Epath%7Bfill:rgb(0,0,0); %7D%3C/style%3E%3C/svg%3E")}.test2{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 263 211' preserveAspectRatio=''%3E%3C/svg%3E")}
```
http://astexplorer.net/#/gist/7a881e5958c9658f933387868d86b17d/bdfdcde57676134bfe1b535914f4db1748251414

# KO when svg with semicolon even on two lines
```css
.test1{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 289 300'%3E%3Cstyle%3Epath%7Bfill:rgb(0,0,0); %7D%3C/style%3E%3C/svg%3E")}
.test2{background-image:url("data:image/svg+xml;utf8,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 263 211' preserveAspectRatio=''%3E%3C/svg%3E")}
```
http://astexplorer.net/#/gist/415b4883c618d16bc716fa5c2ce6423c/ba3bf2c300f2c2c28a16d57083049a20003afcf0

**Comment by pocketjoso** [comment]
Thanks for opening this separate issue with detailed code examples - this is very useful!

You said that the parenthesis issue can be resolved via encoding.. is the same true for the semi colon; can you encode that character as well and does it resolve the problem?

Unfortunately like I said before this issue will only be resolved:
- if the issue is fixed upstream in the [css AST parser](https://github.com/reworkcss/css) (unlikely because neither part of their original design goals, nor maintained)
- if the issue is fixed in [my fork of the above AST Parser](https://github.com/pocketjoso/css) (would require someone to create a PR)

.. so let's keep track of it here until something happens.

**Comment by ryuran** [comment]
> You said that the parenthesis issue can be resolved via encoding.. is the same true for the semi colon; can you encode that character as well and does it resolve the problem?
Yes, if semi-colon and parentheses are encoded it’s working fine.

I update my stylus function to fix this.
https://github.com/ryuran/stylus-svg/blob/master/index.js#L71-L74

**Comment by pocketjoso** [comment]
Okay great, then I'll consider you to be unblocked for now, and I'll look at whetherPenthouse should perhaps try to perform such encoding itself in the future (unless a better AST fix comes up).

**Comment by pocketjoso** [comment]
Should be fixed by `1.2.0` - please give it a go:
https://github.com/pocketjoso/penthouse/releases/tag/v1.2.0
